hashintel/hash
# Only create Python runner when .py behaviors are present

## 🌟 What is the purpose of this PR?

Python and the Python setup should not be required when there is no Python behaviors.
The PR makes them optional.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1202021770319180/f) _(internal)_

## 🔍 What does this change?

- Make the Python runner only spawn when there are Python behaviors present
- Adjust the message loop interacting with the runners to allow non spawned runners
- Update the README

## 📜 Does this require a change to the docs?

- Yes, the README has to be updated as Python is now an optional dependency

## 🛡 What tests cover this?

The tests that don't have any Python behaviors, like the ones in "tests/examples".

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2.  Make sure Python is not setup
3.  Try to run a test that don't have any Python behavior
4.  Confirm that the test pass
